goalie/go_mesh_seq.py

```diff
@@ -185,17 +185,24 @@ def indicate_errors(
                     f", not type '{type(forms)}'."
                 )
 
-            # Loop over each strongly coupled field
-            for f in self.fields:
-                # Loop over each timestep
-                solutions = self.solutions.extract(layout="field")
-                indicators = self.indicators.extract(layout="field")
-                for j in range(len(solutions[f]["forward"][i])):
-                    # Update fields
+            # Loop over each timestep
+            for j in range(self.time_partition.num_exports_per_subinterval[i] - 1):
+                # Loop over each strongly coupled field
+                for f_idx, f in enumerate(self.fields):
+                    # Transfer solutions associated with the current field f
                     transfer(self.solutions[f][FWD][i][j], u[f])
                     transfer(self.solutions[f][FWD_OLD][i][j], u_[f])
                     transfer(self.solutions[f][ADJ][i][j], u_star[f])
                     transfer(self.solutions[f][ADJ_NEXT][i][j], u_star_next[f])
+                    # In case of multiple solution fields, the field that is solved for
+                    # first used the values of other fields from the previous timestep.
+                    # Therefore, transfer the lagged solution of fields f_next that are
+                    # solved for after the current field f. This assumes that the order
+                    # of fields being solved for in get_solver is the same as their
+                    # order in self.fields
+                    if f_idx < len(self.fields) - 1:
+                        for f_next in self.fields[f_idx + 1 :]:
+                            transfer(self.solutions[f_next][FWD_OLD][i][j], u[f_next])
```

Comment:
Is there not repeated effort here? e.g., doesn't the final `FWD_OLD` solution get updated with the same values several times? Could we not just update the `FWD_OLD` solutions for all fields *before* the loop on L191?

Reply:
Ah good point, thanks! Fixed in 1e88b2b0f303b0904fc286c35663774575a55ebc